spec.html

```diff
@@ -4020,6 +4054,220 @@ <h1>CopyDataBlockBytes ( _toBlock_, _toIndex_, _fromBlock_, _fromIndex_, _count_
         </emu-alg>
       </emu-clause>
     </emu-clause>
+
+    <emu-clause id="sec-privatefield-record-specification-type">
+      <h1>The PrivateField Record Specification Type</h1>
+      <p>The PrivateField type is a Record used in the specification of private class fields.</p>
+      <p>Values of the PrivateField type are Record values whose fields are defined as by <emu-xref href="#table-privatefield-fields"></emu-xref>. Such values are referred to as <dfn>PrivateField Record</dfn>s.</p>
+      <emu-table id="table-privatefield-fields" caption="PrivateField Record Fields">
+        <table>
+          <tbody>
+          <tr>
+            <th>
+              Field Name
+            </th>
+            <th>
+              Value
+            </th>
+            <th>
+              Meaning
+            </th>
+          </tr>
+          <tr>
+            <td>
+              [[PrivateFieldName]]
+            </td>
+            <td>
+              a Private Name
+            </td>
+            <td>
+              The name of the field.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[PrivateFieldValue]]
```

Comment:
I don't really like how PrivateField records have a Private Name which has its own `[[Value]]` slot, but ignore that slot and use their own `[[PrivateFieldValue]]` slot. Do we need to structure it this way for some reason? Can't private fields reuse Private Name records?

Reply:
This split, AFAIU, is that private fields have a per-instance value, while private accessors and private methods have a per-class (well, per-brand) value. The per-brand value is kept in the PrivateName's [[Value]] slot, while the fields' per-instance value is kept in [[PrivateFieldValue]].

A rename would be good to anticipate this question, I agree with the current naming this split is not very obvious. Indeed, the branding semantics thing in general is kind of hard to grok.

Reply:
No longer relevant as of https://github.com/tc39/ecma262/pull/1668/commits/558a63cbec0e6d0de22091664de3f17e0e4eccd7.